wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/DeferredCommand.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.wpilibj2.command;
+
+import static edu.wpi.first.util.ErrorMessages.requireNonNullParam;
+
+import edu.wpi.first.util.sendable.SendableBuilder;
+import java.util.Set;
+import java.util.function.Supplier;
+
+/**
+ * Runs the given command when this command is initialized, and ends when it ends. Useful for
+ * performing runtime tasks before creating a new command. If this command is interrupted, it will
+ * cancel the command.
+ *
+ * <p>This class is provided by the NewCommands VendorDep
+ */
+public class DeferredCommand extends Command {
+  private final Supplier<Command> m_supplier;
+  private Command m_command;
+
+  /**
+   * Creates a new DeferredCommand that runs the supplied command when initialized, and ends when it
+   * ends. Useful for lazily creating commands at runtime. The {@link Supplier} will be called each
+   * time this command is initialized. The Supplier <i>must</i> create a new Command each call.
+   *
+   * @param supplier The command supplier
+   * @param requirements The command requirements. This is a {@link Set} to prevent accidental
+   *     omission of command requirements.
+   */
+  public DeferredCommand(Supplier<Command> supplier, Set<Subsystem> requirements) {
+    m_supplier = requireNonNullParam(supplier, "supplier", "DeferredCommand");
+    addRequirements(requirements.toArray(new Subsystem[0]));
```

Comment:
Isn't there a `Iterable<Subsystem>` overload?

Reply:
There is not an overload like that. (Something like that sounded familiar to me too, but the closest thing to an `Iterable<Subsystem>` overload for anything I could find was a `Collection<Command>` overload for `CommandScheduler.requireNotComposed()`)

Reply:
> Isn't there a `Iterable<Subsystem>` overload?

There can be... Should there be?

Reply:
Even if yes, out of scope.